### ISSUE TITLE
"Could not pass event PlayerInteractEvent" on (Spigot) 1.11.2 

### ISSUE THREAD

**Issue by MitesMC** [lishid/OpenInv#60]
We are planning on moving to 1.11.2 and upon randomly testing plugins (and having to open a chest), this error message appeared in the console: https://pastebin.com/0G1LzhtA

Not sure what's causing this as we've never had problems with OpenInv on 1.10.2
Any help would be appreciated. Thank you!

**Comment by Jikoo** [comment]
It appears that OpGuard is injecting a Player implementation other than CraftPlayer. While OpenInv should be checking before casting, OpGuard should have their custom Player implementation extend CraftPlayer to retain compatibility with NMS-based plugins. Due to the changes OpGuard makes, OpenInv will not be able to be used in conjunction with OpGuard at all without a decent bit of special handling.

I will include a check before the cast and a warning in console, but I have no plans to special case plugins which alter standard behavior.

Just my two cents: You should not be running plugins you have not examined or do not trust. A plugin like OpGuard is relatively easily circumvented and only creates a hassle for legitimate developers that need deeper access. For example, if I were planning on making a force-op style plugin, I wouldn't grant myself op at all, I'd just include commands that let me set my own permissions.

**Comment by MitesMC** [comment]
@Jikoo thank you for your insight. This is actually the first time that I've used OpGuard. And since the 1.11.2 version of our server is still under works, I've added that plugin to the mix, mainly to try it out and test its effectivity. But with this conflict arising with OpenInv, it's highly possible that it may happen to other plugins as well, and I have yet to discover that conflict, I guess we're better off without it. Thank you again for your assistance.
